python/GafferUI/Slider.py

```diff
@@ -192,48 +217,92 @@ def getMinimumSize( self ) :
 
 		return self.__minimumSize
 
-	## Sets the size of the position increment added/subtracted
+	## Sets the value increment added/subtracted
 	# when using the cursor keys. The default value of None
 	# uses an increment equivalent to the size of one pixel at
 	# the current slider size. An increment of 0 can be specified
 	# to disable the behaviour entirely.
-	## \todo Add setValueIncrement() method on NumericSlider.
-	def setPositionIncrement( self, increment ) :
+	def setIncrement( self, increment ) :
 
-		self.__positionIncrement = increment
+		self.__increment = increment
 
-	def getPositionIncrement( self ) :
+	def getIncrement( self ) :
 
-		return self.__positionIncrement
+		return self.__increment
 
-	## May be overridden by derived classes if necessary, but
-	# implementations must call the base class implementation
-	# after performing their own work, as the base class is
-	# responsible for emitting positionChangedSignal().
-	def _setPositionsInternal( self, positions, reason ) :
+	def __setValuesInternal( self, values, reason ) :
 
-		dragBeginOrEnd = reason in ( self.PositionChangedReason.DragBegin, self.PositionChangedReason.DragEnd )
-		if positions == self.__positions and not dragBeginOrEnd :
-			# early out if the positions haven't changed, but not if the
+		# We _always_ clamp to the hard min and max, as those are not optional.
+		# Optional clamping to soft min and max is performed before calling this
+		# function.
+		values = [ max( self.__hardMin, min( self.__hardMax, x ) ) for x in values ]
+
+		dragBeginOrEnd = reason in ( self.ValueChangedReason.DragBegin, self.ValueChangedReason.DragEnd )
+		if values == self.__values and not dragBeginOrEnd :
+			# early out if the values haven't changed, but not if the
 			# reason is either end of a drag - we always signal those so
 			# that they will always come in matching pairs.
 			return
 
-		self.__positions = positions
+		self.__values = values
 		self._qtWidget().update()
 
-		self.__emitPositionChanged( reason )
+		self.__emitValueChanged( reason )
 
 	## \todo Colours should come from some unified style somewhere
 	def _drawBackground( self, painter ) :
 
 		size = self.size()
+		valueRange = self.__max - self.__min
+		if valueRange == 0 :
+			return
 
-		pen = QtGui.QPen( QtGui.QColor( 0, 0, 0 ) )
-		pen.setWidth( 1 )
-		painter.setPen( pen )
+		idealSpacing = 10
+		idealNumTicks = float( size.x ) / idealSpacing
+		tickStep = valueRange / idealNumTicks
+
+		logTickStep = math.log10( tickStep )
+		flooredLogTickStep = math.floor( logTickStep )
+		tickStep = math.pow( 10, flooredLogTickStep )
+		blend = (logTickStep - flooredLogTickStep)
+
+		tickValue = math.floor( self.__min / tickStep ) * tickStep
+		i = 0
+		while tickValue <= self.__max :
```

Comment:
We may need a way to turn ticks off?

OLD
![image](https://user-images.githubusercontent.com/896779/104332821-7f4f9580-54e8-11eb-9f66-2d33930f65f2.png)
NEW

Reply:
We chatted about this separately and decided we didn't mind the ticks too much in the end.